zcash_primitives/src/transaction/mod.rs

```diff
@@ -725,16 +728,13 @@ impl Transaction {
     }
 
     fn read_v5_header_fragment<R: Read>(mut reader: R) -> io::Result<(BranchId, u32, BlockHeight)> {
-        let consensus_branch_id = reader.read_u32::<LittleEndian>().and_then(|value| {
-            BranchId::try_from(value).map_err(|e| {
-                io::Error::new(
-                    io::ErrorKind::InvalidInput,
-                    "invalid consensus branch id: ".to_owned() + e,
-                )
+        let consensus_branch_id = reader.read_u32_le().and_then(|value| {
+            BranchId::try_from(value).map_err(|_e| {
+                io::Error::new(io::ErrorKind::InvalidInput, "invalid consensus branch id")
```

Comment:
This one is particularly irritating, since converting a consensus branch id to a string should not even require allocation.

Reply:
Actually this is not a regression because `BranchId::try_from` only returns an error with the fixed message "Unknown consensus branch ID". If we want to include the invalid id in the message, this works in no_std:

```suggestion
                let mut msg: [u8; 38] = *b"invalid consensus branch id 0x........";
                (&mut msg[30..38]).copy_from_slice(hex::encode(value.to_be_bytes()).as_bytes());
                io::Error::new(
                    io::ErrorKind::InvalidInput,
                    core::str::from_utf8(&msg).expect("valid utf8"),
                )
```
Non-blocking.

Reply:
This still doesn't work because the argument must have `'static` lifetime.